packages/gestalt/src/TextField/InternalTextField.js

```diff
@@ -213,17 +221,26 @@ const InternalTextFieldWithForwardRef: AbstractComponent<Props, HTMLInputElement
     />
   );
 
+  // Explicit margin for the small size, we don't have a token for 2px
+  const tagMarginY = size === 'sm' || size === 'md' ? '2px' : 'var(--space-100)';
+
   return (
     <span>
-      {label ? <FormLabel id={id} label={label} labelDisplay={labelDisplay} /> : null}
+      {label ? <FormLabel id={id} label={label} labelDisplay={labelDisplay} size={size} /> : null}
 
       <Box position="relative">
         {tags ? (
           <div className={styledClasses} {...(tags ? { ref: innerRef } : {})}>
             {tags.map((tag, tagIndex) => (
-              // eslint-disable-next-line react/no-array-index-key
-              <Box key={tagIndex} marginEnd={1} marginBottom={1}>
-                {tag}
+              <Box
+                // eslint-disable-next-line react/no-array-index-key
+                key={tagIndex}
+                marginEnd={1}
+                dangerouslySetInlineStyle={{
+                  __style: { marginBottom: tagMarginY },
```

Comment:
nit we could inline the logic

Reply:
We can't sadly. In order to get the sizing right, I had to set the tagMargin to 2px, that's outside of our usual scale